templates/user/user-about.html

```diff
@@ -67,10 +130,122 @@ <h4>Rating History</h4>
+                    link.hover(
+                        function(e) {
+                            display_tooltip(link);
+                        },
+                        function(e) {
+                            display_tooltip(null);
+                        }
+                    );
+                });
+            }
+
+            var current_year = new Date().getFullYear();
+            var $div = $('#submission-activity');
+
+            function draw_contribution(year) {
+                $div.find('#submission-activity-table td').remove();
+                $div.find('#year').attr('data-year', year);
+                $div.find('#prev-year-action').css('display', year > metadata.min_year ? '' : 'none');
+                $div.find('#next-year-action').css('display', year < current_year ? '' : 'none');
+                var current_weekday = 0;
+                var start_day = new Date(year, 0, 1)
+                var end_day = new Date(year + 1, 0, 0);
+                if (year == current_year) {
+                    end_day = new Date();
+                    start_day = new Date(end_day.getFullYear() - 1, end_day.getMonth(), end_day.getDate() + 1);
+                    $div.find('#year').text("{{ _('past year') }}");
+                } else {
+                    $div.find('#year').text(year);
+                }
+                var days = [];
+                for (var day = start_day, day_num = 1; day <= end_day; day.setDate(day.getDate() + 1), day_num++) {
+                    var isodate = day.toISOString().split('T')[0];
+                    days.push({
+                        date: new Date(day),
+                        weekday: day.getDay(),
+                        day_num: day_num,
+                        activity: submission_activity[isodate] || 0,
+                    });
+                }
+
+                var sum_activity = days.map(obj => obj.activity).reduce((a, b) => a + b, 0);
+                $div.find('#submission-total-count').text(
+                    ngettext("%(cnt)d total submission", "%(cnt)d total submissions", sum_activity)
+                        .replace("%(cnt)d", sum_activity)
+                )
+
+                var max_activity = Math.max.apply(null, days.map(obj => obj.activity));
+                var diff = max_activity / (activity_levels - 1);
+                var activity_breakdown = [...Array(activity_levels).keys()].map(x => diff * x);
+
+                for (; current_weekday < days[0].weekday; current_weekday++) {
+                    $div.find('#submission-' + current_weekday)
+                        .append($('<td>').addClass('activity-blank').append('<div>'));
+                }
+
+                days.forEach(obj => {
+                    var level = activity_breakdown.findIndex(x => x >= obj.activity);
+                    var text =
+                        ngettext("%(cnt)d submission on %b %d, %Y", "%(cnt)d submissions on %b %d, %Y", obj.activity)
```

Comment:
This is terrible i18n for the date. While you can rearrange the symbols, not every language uses full month names. For example, it would be ridiculous to show `2020年八月10日` instead of something more sane like `2020年8月10日`, which is the best you can do with this code.

Reply:
Oh and I forgot, in some languages, the month name needs to be in the genitive case for dates. Your nominative month names won't cut it.

Reply:
From some discussion on slack, there are a few options (since there doesn't seem to be anything in the Django docs for formatting dates in JavaScript):
 1. Format the activity date thing on the server.
      - I'm not sure how this will work, since wouldn't we need to format _all_ date strings (which could grow very large if the user has been active for multiple years)?
 2. Use the user's browser, and using something like `date.toLocaleDateString()` (or maybe moment.js's `format`?).
    - `date.toLocaleDateString()` could work, but it returns something like `10/08/2020`, which is tolerable, but not ideal for users.
    - moment.js's `format` - We'll need a way to convert from Django's date formatting to moment.js's format if we want to keep the date formatting consistent (using `get_format("DATE_FORMAT")`).

Reply:
Option 3: just use ISO date formats, which are internationally recognized.

Reply:
I found another option:

Would something like `new Date().toLocaleDateString("{{ LANGUAGE_CODE }}", {month: 'short', year: 'numeric', day: 'numeric'})` work? This will format dates to `2020年8月11日`, `11 Aug 2020`, and ` авг. 2020 г.`.

Reply:
Looks about right. You can use that. The string you quoted for Russian is wrong, but it does not appear to be a bug in the code.

Reply:
Done